src/core/TotpCore.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+* Copyright (c) 2003-2023 Rony Shapiro <ronys@pwsafe.org>.
+* All rights reserved. Use of the code is allowed under the
+* Artistic License 2.0 terms, as specified in the LICENSE file
+* distributed with this code, or available from
+* http://www.opensource.org/licenses/artistic-license-2.0.php
+* Contributed by Ashley R. Thomas, 18-Oct-2023
+*/
+// TotpCore.cpp
+//-----------------------------------------------------------------------------
+
+/**
+ * Password Safe Core library two factor authentication related structs and
+ * funtionality connecting PwSafe Core users (i.e., UI/CLI) with TOTP internals.
+ */
+
+#include <string>
+#include <sstream>
+#include "core/crypto/totp.h"
+#include "UTF8Conv.h"
+#include "ItemData.h"
+#include "TotpCore.h"
+
+namespace PWSTotp {
+
+  std::wstring GetTotpErrorString(TOTP_Result r)
+  {
+    switch (r) {
```

Comment:
All these strings should be resources.
Also, they should be rephrased to me more actionable by the user. For example, the typical user has no clue as to what 'base32' or 'UTF8' mean. A more useful message would be something like "Key can contain only the following letters: ABCDEFGHIJKLMNOPQRSTUVWXYZ234567 and must be ?? characters long"